File: scripts/Symmetric Tree/Symmetric Tree.py
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        return root is None or self.findSymmetric(root.left, root.right)
    
    def findSymmetric(self, left, right):
        if (left is None or right is None):
            return left == right
        
        if (left.val != right.val):
            return False
        
        return self.findSymmetric(left.left, right.right) and self.findSymmetric(left.right, right.left)

File: scripts/Symmetric Tree/Symmetric Tree.cpp
class Solution {
public:
    
    bool mirror(TreeNode* root1, TreeNode* root2){
        if(root1==NULL and root2==NULL) return true;
        if(root1==NULL or root2==NULL) return false;
        
        if(root1->val!=root2->val) return false;
        
        return mirror(root1->left, root2->right) and mirror(root1->right,root2->left);
    }
    
    bool isSymmetric(TreeNode* root) {
        if(root==NULL) return true;
        
        return mirror(root->left,root->right);
    }
};
